types/trigger.d.ts

```diff
@@ -74,56 +86,58 @@ export type TriggerAutoConfig = {
   SpokenAlertsEnabled?: boolean;
 }
 
-export type MatchesAny = { [s in T]?: string };
-
 // Note: functions like run or preRun need to be defined as void-only as (confusingly)
 // it is not possible to assign `(d: Data) => boolean` to a void | undefined, only to void.
-export type TriggerField<Data, Return> =
-  [Return] extends [void] ? TriggerFunc<Data, MatchesAny, void> :
-  TriggerFunc<Data, MatchesAny, Return | undefined> | Return | undefined;
+export type TriggerField<Data, Return, Matches> =
```

Comment:
I am sorry for the most obnoxious nit here, but can these be ordered left to right like they'd be in a trigger? e.g. ?<data, matches, (output), return>?